### PR TITLE
Reworking geometry property names a little

### PR DESCRIPTION
**Pull request by jeherr**
Renaming properties to labels because it is shorter to type. Also starting an atom object.

Additionally this changed the make_dataset to use the JSON library directly, which should simplify thing a bit.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
     include_package_data=True,
 
     # Allows `setup.py test` to work correctly with pytest
-    setup_requires=[] + pytest_runner, install_requires=['torch', 'numpy']
+    setup_requires=[] + pytest_runner, install_requires=['torch', 'numpy', 'ase']
```
Comment (alexisdriscoll) [inline setup.py:44]: Thanks I forgot to do this

File: tensorchem/dataset/molecule.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 """
 
 import json
 import torch
 
 
 class MoleculeSet:
-    def __init__(self, atomic_nums=None):
-        self.atomic_nums = ()
+    def __init__(self, atomic_nums=(0,)):
```
Comment (alexisdriscoll) [inline tensorchem/dataset/molecule.py:11]: Why are you adding the 0 and not just making it blank?

File: tensorchem/dataset/molecule.py [hunk 3]
```diff
@@ -54,15 +54,15 @@ def load(self, filename=None):
...
         self.atomic_nums = atomic_nums
         self.coords = coords
-        self.properties = properties
+        self.labels = labels
 
     def __repr__(self):
-        return str(self.export_json())
-
-    def __str__(self):
+        # TODO add some of that fancy printing crap to make this print the xyz in a nicer format
```
Comment (alexisdriscoll) [inline tensorchem/dataset/molecule.py:65]: Haha

Comment (alexisdriscoll) [reply tensorchem/dataset/molecule.py:65]: I can try to work on this

File: tensorchem/dataset/molecule.py [hunk 4]
```diff
@@ -91,6 +90,7 @@ def __get_dist_matrix__(self):
         return torch.norm(self.coords)
 
 
-if __name__ == "__main__":
-    mol1 = MoleculeSet()
-    mol1.load('19021.mset')
+class Atom:
```
Comment (alexisdriscoll) [inline tensorchem/dataset/molecule.py:93]: Where do you want to use this?

File: tensorchem/util/make_dataset.py [hunk 5]
```diff
@@ -25,78 +25,90 @@
...
-                f2.write('[' + str(coords[i][y][0]) + ', ' + str(coords[i][y][1]) + ', ' + str(coords[i][y][2]) + ']]], ')
-            else:
-                f2.write('[' + str(coords[i][y][0]) + ', ' + str(coords[i][y][1]) + ', ' + str(coords[i][y][2]) + ']], ')
-        f2.write('"properties": {"energies": ' + str(float(energies[x])) + ', "forces": ' + str(forces[x]).replace("'", "") + ', "dipoles": '\
-            + str(dipoles[x]).replace("'", "") + ', "quadrupoles": ' + str(quadrupoles[x]).replace("'", "") + ', "charges": ' + str(charges[x]).replace("'", "") + '}}')
-        f2.close()
+    if converged:
+        if len(atomic_nums) == len(coords) == len(energies) == len(dipoles) == len(quadrupoles) == len(charges) > 0:
+            for i, atomic_num_1 in enumerate(atomic_nums):
+                for atomic_num_2 in atomic_nums[i+1:]:
```
Comment (alexisdriscoll) [inline tensorchem/util/make_dataset.py:92]: Is this going to fail when you reach the last indexing point in atomic_nums?

File: tensorchem/util/make_dataset.py [hunk 6]
```diff
@@ -25,78 +25,90 @@
...
+                    if tuple(atomic_num_1) != tuple(atomic_num_2):
+                        print("Atomic numbers are not the same between frames")
+                        print(mol)
+                        exit(0)
+            mset_data = {"atomic_numbers": atomic_nums[len(atomic_nums) - 1]}
+            geometries = [{"coordinates": coord} for coord in coords]
+            for i, geom in enumerate(geometries):
+                geom.update({"labels": {"wb97x-d/6-311gss.energy": energies[i],
+                             "wb97x-d.6-311gss.forces": forces[i],
```
Comment (alexisdriscoll) [inline tensorchem/util/make_dataset.py:101]: Why are you using this specific label for the energy, forces, etc.

File: tensorchem/util/make_dataset.py [hunk 7]
```diff
@@ -25,78 +25,90 @@
...
+                             "wb97x-d.6-311gss.dipole": dipoles[i],
+                             "wb97x-d.6-311gss.quadrupole": quadrupoles[i],
+                             "wb97x-d.6-311gss.mulliken_charges": charges[i]}})
+            mset_data.update({"geometries": geometries,
```
Comment (alexisdriscoll) [inline tensorchem/util/make_dataset.py:105]: This implementation is much cleaner and makes much more sense. Thanks for fixing it up for me. This is definitely more readable too

File: tensorchem/util/make_dataset.py [hunk 8]
```diff
@@ -25,78 +25,90 @@
...
+                              "identifiers": {"chemspider_id": mol[:-4]}})
+            with open(os.path.join("/Users/johnherr/tensorchem/data/", ".".join((mol[:-4], "mset"))), "w") as f2:
+                json.dump(mset_data, f2)
+        else:
+            print(mol)
+            print(len(atomic_nums), len(energies), len(forces), len(dipoles), len(quadrupoles), len(charges))
+            exit(0)
     else:
-        print(mol)
+        print("Optimization did not converge for "+mol)
```
Comment (alexisdriscoll) [inline tensorchem/util/make_dataset.py:114]: Good addition

File: tensorchem/dataset/molecule.py (hunk 2 above)
Comment (jeherr) [reply tensorchem/dataset/molecule.py:11]: It gives it a base value with the same type as what we expect the variable should be. So if it is set to None, then it throws lots of Nonetype errors. Instead, we can see obviously if the atoms are set to (0,) then they were never set actually, they just kept the minimal initialization value. It makes it more obvious where the error comes from. It makes for a nice sort of bread crumb to finding errors, and it also gives a type suggestion. This suggests using tuples to set the atomic numbers (though I am thinking of changing that) instead of lists, because tuples are immutable (the order cannot change).

File: tensorchem/dataset/molecule.py (hunk 4 above)
Comment (jeherr) [reply tensorchem/dataset/molecule.py:93]: I'm thinking of changing the MoleculeSet a little to have atoms objects as well. So the MoleculeSet could have a list of Atom objects, which then just get filled with coordinates to create the geometry. The reason is it allows for a little more flexibility in how the data is stored. 

We're generally working in 3D coordinates, so the order of atoms doesn't matter. But if we store everything in structured lists, then we have to maintain the order, otherwise we might mix up the forces of one atom with the coordinates of another. If an atom holds its own forces and coordinates, then if the atoms get shuffled, they still keep their correct data because it is shuffled with the atom.

I'll add some more to this soon to get it worked in.

File: tensorchem/util/make_dataset.py (hunk 5 above)
Comment (jeherr) [reply tensorchem/util/make_dataset.py:92]: Nah it just 'iterates' over an empty list, so it happens completely gracefully. No errors. Already used it to convert a lot of the data from one of the archive files as a test.

File: tensorchem/util/make_dataset.py (hunk 6 above)
Comment (jeherr) [reply tensorchem/util/make_dataset.py:101]: This is the functional and basis set we used to calculate this data. The functional is wB97X-D, and the basis set is 6-311G**. I think I meant to use a "." to join each bit of information together but messed it up a bit here. Either way I am intending the format to be like "functional.basis.property".
